apps/greencheck/templates/add_latest_check_for_url_procedure.sql

```diff
@@ -0,0 +1,81 @@
+DROP PROCEDURE IF EXISTS add_latest_check_for_url;
+
+-- define our procedure, that accepts the url param, as a VarChar
+CREATE PROCEDURE add_latest_check_for_url(IN purl VarChar(255))
+
+BEGIN
+
+
+  -- declare the types we'd fetch from the cursor
+  DECLARE g_id            BIGINT(20);
+  DECLARE g_id_hp         INT(11);
+  DECLARE g_id_greencheck INT(11);
+  DECLARE g_type          ENUM('url','whois','ip','none','as');
+  DECLARE g_url           VARCHAR(255);
+  DECLARE g_ip            DECIMAL(39,0);
+  DECLARE g_datum         TIMESTAMP;
+  DECLARE g_green         ENUM('yes','no','old');
+  DECLARE g_tld           VARCHAR(64);
+
+  -- declare the other types for the next query
+  DECLARE h_hostname        VARCHAR(255);
+  DECLARE h_hostpartner     VARCHAR(255);
+  DECLARE h_hostwebsite     VARCHAR(255);
+
+
+  -- SELECT concat('** checking for: ', purl) AS '** DEBUG:';
+
+  DECLARE CONTINUE HANDLER FOR NOT FOUND
+    BEGIN
+      -- SELECT concat('** NO RESULT FOR: ', purl) AS '** DEBUG:';
+      -- SELECT concat('** BUT WE SHOULD NOT BUBBLE THE EXCEPTION UP: ', '') AS '** DEBUG:';
+    END;
```

Comment:
What does the following five lines of code do?

Reply:
In what way is this connected to line 23 in backfill_by_url_procedure?

Reply:
As far as I can tell, stored procedures bubble `NOT FOUND` results up to the calling stored procedure.

So the `backfill_by_url` procedure was dropping out of the loop early, as soon as a for a url didn't return any results.

This meant that green presenting wasn't populated at all if I didn't catch it.